hack/build-images.sh

```diff
@@ -78,39 +72,78 @@ cp -pf "${OS_ROOT}/contrib/systemd/openshift-sdn-ovs.conf" images/node/conf/
 
 # builds an image and tags it two ways - with latest, and with the release tag
 function image {
-  local STARTTIME=$(date +%s)
-  echo "--- $1 ---"
-  build $1:latest $2
-  #docker build -t $1:latest $2
-  docker tag $1:latest $1:${OS_RELEASE_COMMIT}
-  git clean -fdx $2
-  local ENDTIME=$(date +%s); echo "--- $1 took $(($ENDTIME - $STARTTIME)) seconds ---"
-  echo
-  echo
+  local tag=$1
+  local dir=$2
+  local out
+  out="$( mktemp )"
+
+  set +e
+  (
+    set -euo pipefail
+
+    local dest="${tag}"
+    if [[ ! "${tag}" == *":"* ]]; then
+      dest="${tag}:latest"
+    fi
+
+    local STARTTIME=$(date +%s)
+
+    # build the image
+    if ! os::build::image "${dir}" "${dest}"; then
+      os::log::warn "Retrying build once"
+      os::build::image "${dir}" "${dest}"
+    fi
+
+    # tag to release commit unless we specified a hardcoded tag
+    if [[ ! "${tag}" == *":"* ]]; then
+      docker tag "${dest}" "${tag}:${OS_RELEASE_COMMIT}"
+    fi
+    # ensure the temporary contents are cleaned up
+    git clean -fdx "${dir}"
+
+    local ENDTIME=$(date +%s); echo "Finished in $(($ENDTIME - $STARTTIME)) seconds"
+  ) > "${out}" 2>&1
+  local rc=$?
+  set -e
+
+  if [[ "${rc}" -ne 0 ]]; then
+    sed -e "s|^|$1: |" "${out}" 1>&2
+    os::log::error "Failed to build $1"
+  else
+    sed -e "s|^|$1: |" "${out}"
+  fi
+  return $rc
 }
 
 # images that depend on scratch / centos
-image openshift/origin-pod                   images/pod
-image openshift/openvswitch                  images/openvswitch
+( image openshift/origin-pod                   images/pod ) &
```

Comment:
Is `oc ex dockerbuild` changing the environment somehow? You can parallelize without the subshell.